src/drivers/imx/sdma.c

```diff
@@ -0,0 +1,848 @@
+#include <sof/audio/component.h>
+#include <sof/drivers/sdma.h>
+#include <sof/drivers/timer.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/dma.h>
+#include <sof/lib/io.h>
+#include <sof/lib/notifier.h>
+#include <sof/lib/wait.h>
+#include <sof/platform.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define trace_sdma(fmt, ...) trace_event(TRACE_CLASS_DMA, fmt, ##__VA_ARGS__)
+#define tracev_sdma(fmt, ...) tracev_event(TRACE_CLASS_DMA, fmt, ##__VA_ARGS__)
+#define trace_sdma_error(fmt, ...) trace_error(TRACE_CLASS_DMA, fmt, ##__VA_ARGS__)
+
+#define SDMA_BUFFER_PERIOD_COUNT 2
+
+struct sdma_bd {
+	/* SDMA BD configuration */
+	uint32_t config;
+
+	/* Buffer addresses, typically source and destination in some
+	 * order, dependent on script
+	 */
+	uint32_t buf_addr;
+	uint32_t buf_xaddr;
+} __packed;
+
+struct sdma_context {
+	uint32_t pc;
+	uint32_t spc;
+	uint32_t g_reg[8];
+	uint32_t dma_xfer_regs[14];
+	uint32_t scratch[8];
+} __packed;
+
+struct sdma_ccb {
+	uint32_t current_bd_paddr;
+	uint32_t base_bd_paddr;
+	uint32_t status;
+	uint32_t reserved; /* No channel descriptor implemented */
+};
+
+/* This structure includes all SDMA related channel data. */
+struct sdma_chan {
+	struct sdma_bd descriptors[SDMA_MAX_BDS];
+	int descriptor_count;
+	struct sdma_context *ctx;
+	struct sdma_ccb *ccb;
+	int hw_event;
+	int sdma_chan_type;
+	int fifo_paddr;
+};
+
+
+/* Private data for the whole controller */
+struct sdma_pdata {
+	struct sdma_chan *chan_pdata;
+	struct sdma_context *contexts;
+	struct sdma_ccb *ccb_array;
+};
+
+static void sdma_set_overrides(struct dma_chan_data *channel, bool event_override, bool host_override) {
+	trace_sdma_error("sdma_set_overrides(%d, %d)", event_override, host_override);
+	dma_reg_update_bits(channel->dma, SDMA_EVTOVR, BIT(channel->index), event_override ? BIT(channel->index) : 0);
+	dma_reg_update_bits(channel->dma, SDMA_HOSTOVR, BIT(channel->index), host_override ? BIT(channel->index) : 0);
+}
```

Comment:
I think this line goes over 80 chars. In fact I think there are several places where this happens. Maybe run checkpatch.ppl with --strict?

Reply:
Will fix the lines-too-long issues as I look again into this file. I still have a few dummy functions that need to go.

Reply:
thanks.